Medical Additions / Corpsman Job

KillerTheFlareon opened the pull request:
<!-- IT'S NOT WIZDENS REPO, IF YOU WANT TO ADD YOUR CHANGES ON ALL SERVERS, CREATE PR TO WIZDENS REPO -->

## Short description
<!-- What do you propose to change with your PR? -->
Adds new items to medical, and reworks corpsman into a new job.
## Why we need to add this
<!-- What is the reason for adding these changes? Please post links to Discussions as well as Bug Reports here. Please describe how this will change the game balance. -->
Medical does not like the entire chemlocker going boom. This gives them an alternative that they can make with reinforced glass as a T2 med research (vials/bottles/jugs/beakers) which also boast a nice boost to capacity, Para gets a portable defib like the corps and i've made a new item called "hypoinjector" which is essentially a faster syringe themed as an autoinjector with a limiter (press injector against skin, hold down button to draw or press button to inject), which I'm sure med will enjoy.

I've fixed brig loadouts so all their gear now spawns properly in the locker, and also separated brig's duties into 2 roles instead of 1 due to the increased demand on highpop. There is now a new corpsman role that handles all the external duties of brig (rescuing crit/dead security officers, providing on-field medical treatment & responding to calls for medical attention over coms)

This is mainly due to the fact the brig was expected to act like a doctor/chemist/paramedic/combat med all at once, which works well enough in lower pop but results in "the brig is never in the brig" on higher pop as they're constantly running around rescuing officers or providing immediate medical attention in the field whilst prisoners die in the brig. Or officers show up to be treated only to have no brigmedic in brig.

So Corpsman takes on the para/combat med responsibilities which takes weight off of the brigmedic's shoulders allowing them to focus on prisoners/seccies in sec or just cooking chems, and also allows them to follow their sop of "you really shouldn't leave brig as brigmedic" which was previously impossible to follow due to the contradicting responsibilities. 

This also means all of the previous mobility tools (minus the brig monitor, as I don't expect mappers to start mapping in a vitals monitor to brigmed anytime soon) are removed from brig and given to corps instead (defib/big medkit & a new 10u vial using hypospray, which is also a syndicate target objective.) 

I've also made the brigmedic bedsheet a thief target, like it was intended to be but never done.
## Media (Video/Screenshots)
<!--
If your PR contains in-game changes you must provide screenshots/videos of the changes.
-->
Corp & Brig side-by-side
<img width="652" height="476" alt="ytjuytfdk" src="https://github.com/user-attachments/assets/8d442085-dc7c-48d0-ad46-da386b59917d" />
It has a hood!
<img width="581" height="446" alt="tyjdutgk" src="https://github.com/user-attachments/assets/4a11b896-b0c5-485f-8505-b55c6dd121b1" />
Medical additions
<img width="497" height="360" alt="tyjddtuk" src="https://github.com/user-attachments/assets/71988e0c-a631-466d-9dd2-d89f4fa5a4c9" />

## Checks
<!-- check boxes for faster reviewing of your PR -->

- [x] I do not require assistance to complete the PR.
- [x] Before posting/requesting review of a PR, I have verified that the changes work.
- [x] I have added screenshots/videos of the changes, or this PR does not change in-game mechanics.
- [x] I affirm that my changes are licensed under the [Starlight Fork License](https://github.com/ss14Starlight/space-station-14/blob/Starlight/LICENSE-Starlight.TXT) and grant permission for use in this repository under its conditions.

**Changelog**
<!--
If you want the players to know about changes made in this PR, specify them using the template outside the comment. Short and informative.
-->
:cl: Killer Tamashi (The rest) & Faush (C# Code) & Salem (Brigmedic mask sprite)
- add: Added Reinforced Containers (T2 Med) [Reinforced Vial/Bottle/Small Beaker/Large Beaker/Jug]
- add: Added Automated Injection (T1 Med) [Hypoinjector, literally just a faster syringe]
- add: Corpsman Job & Items [Corps everything basically & Hypospray Kit]
- add: Corpsman Hypospray [Uses 10u vials, syndie objective]
- add: Corpsman to all highpop maps & other where applicable
- add: Secmed Key, you can guess whose headset it goes into
- tweak: Renamed Brigmedic Hardsuit & Headset into "Security Medical Suit/Headset" respectively
- tweak: Resprited Brigmedic Headset into a grey/blue colorscheme to more fit in with sec headsets
- tweak: Brigmedic mask now has 5% infection resist chance
- tweak: Brigmedic jacket now has 40% infection resist chance, on par with winter coats.. Why do winter coats prevent-
- tweak: Resprited Brigmedic Icon to match with new Corps icon
- tweak: Brigmedic mask resprited to look better.
- fix: Corpsman jacket looking AWFUL
- fix: SOMEONE butchered the brigmedic locker's fills, this also fixes that.

KillerTheFlareon committed:
Add brigmed hypo+case+fix

..magboot spawn chance on brigmedic locker. Someone accidentally gave it a 0.1 chance to spawn lol.

KillerTheFlareon committed:
Merge branch 'Starlight' into Brigmed-Hypo

KillerTheFlareon committed:
Round 2

KillerTheFlareon committed:
Reinforced stuff

KillerTheFlareon committed:
R.Glass recipes

You can now put reinforced glass in ANY lathe, but only the medical one has any recipes for them

PubliclyExecutedPig commented:
make sure to talk to grape for the sop stuff surrounding the hypo

Dimyell commented:
Having the hypospray also need a vial is a really cool idea actually. Makes it both balanced for those who think brigmedic having a hypospray would break balance and also gives vials more use cases.

Fasuh committed:
feat(Chemistry): Add logic for miniature hypospray to use inserted vials in place of caintainer.

KillerTheFlareon committed:
awawa

aaa

KillerTheFlareon committed:
Brigmedic clothing resprite & Hypoinjector sprites

Fixed geneva convention violation
Also gave the hypo a sprite so yippee

Fasuh committed:
fix(Hypospray): Remove drawing blood and from containers from brigmedic hypospray

Fasuh committed:
fix(Hypoinjector): add translations for hypoinjector

KillerTheFlareon committed:
a

a

KillerTheFlareon committed:
Adds steal objectives

- Brig bedsheet was always intended to be one, i just.
Didn't

- Hypo is syndie steal objective
self explanatory i'm sure.

KillerTheFlareon committed:
Forgot to change brigmedic defib

As well as FINALLY added para defib

KillerTheFlareon committed:
aaaaa

KillerTheFlareon committed:
Merge branch 'Starlight' into Brigmed-Hypo

KillerTheFlareon committed:
Corpsman Addition

Whew. That was a lot of work.
This finally makes a role separation between corpsman and brigmedic, allowing the brigmedic to stay in brig and the corpsman to run out and perform the other half of the brigmedic's duties.

Basically, brig is doctor/chemist, stuck in brig
Corps is para/on field triage meant to retrieve & heal seccies in combat conditions

KillerTheFlareon commented:
This used to be a simple pr. Used.

KillerTheFlareon committed:
Adds missing things.

Ye.

KillerTheFlareon committed:
Fixes

wawa

KillerTheFlareon committed:
Merge branch 'Starlight' into Brigmed-Hypo

KillerTheFlareon committed:
Merge branch 'Starlight' into Brigmed-Hypo

KillerTheFlareon committed:
Delete .equipped-OUTERCLOTHING.png-autosave.kra

KillerTheFlareon commented:
As I mentioned to happy, whom I know will be reviewing this PR, I cannot actually TEST if corpsman spawns properly because my dev environment is a piece of shit that doesn't work half the time. Therefore I am requesting, before this is merged, that happy tries to spawn as a corpsman on any station during whatever testing he does.

KillerTheFlareon committed:
Update chemistry-vials.yml

PubliclyExecutedPig commented:
> As I mentioned to happy, whom I know will be reviewing this PR, I cannot actually TEST if corpsman spawns properly because my dev environment is a piece of shit that doesn't work half the time. Therefore I am requesting, before this is merged, that happy tries to spawn as a corpsman on any station during whatever testing he does.

Just to ask, have you used the "runserver.bat RELEASE" version or is your dev like truly fucked

KillerTheFlareon commented:
> > As I mentioned to happy, whom I know will be reviewing this PR, I cannot actually TEST if corpsman spawns properly because my dev environment is a piece of shit that doesn't work half the time. Therefore I am requesting, before this is merged, that happy tries to spawn as a corpsman on any station during whatever testing he does.
> 
> Just to ask, have you used the "runserver.bat RELEASE" version or is your dev like truly fucked

I use the "runclient" / "runserver" .bats
Never use any different ones. My dev is just fucked for some reason

KillerTheFlareon commented:
OKAY SO I RAN TOOLS AND THAT WORKED.
Corpsman spawns fine, so don't need happy to check-

Well. "Fine" - They still need spawnpoints mapped, but, you know they exist in round.

KillerTheFlareon committed:
Update medical.yml

I forsee great squabbling over this if I didn't rename it

PubliclyExecutedPig commented:
> OKAY SO I RAN TOOLS AND THAT WORKED. Corpsman spawns fine, so don't need happy to check-
> 
> Well. "Fine" - They still need spawnpoints mapped, but, you know they exist in round.

ya runserver.bat is just for quick testing on items and such, for actual testing i recommend running 
<img width="207" height="45" alt="image" src="https://github.com/user-attachments/assets/71083cf5-c9cc-460e-a25b-5f1b9768b87e" />
this file, its basically just the full build

KillerTheFlareon commented:
> > OKAY SO I RAN TOOLS AND THAT WORKED. Corpsman spawns fine, so don't need happy to check-
> > Well. "Fine" - They still need spawnpoints mapped, but, you know they exist in round.
> 
> ya runserver.bat is just for quick testing on items and such, for actual testing i recommend running <img alt="image" width="207" height="45" src="https://private-user-images.githubusercontent.com/152017811/468275089-71083cf5-c9cc-460e-a25b-5f1b9768b87e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI5MTY2ODgsIm5iZiI6MTc1MjkxNjM4OCwicGF0aCI6Ii8xNTIwMTc4MTEvNDY4Mjc1MDg5LTcxMDgzY2Y1LWM5Y2MtNDYwZS1hMjViLTVmMWI5NzY4Yjg3ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcxOVQwOTEzMDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNzY3NzQ0Njk1YWM1NjI2NjFiMmNjYTM1NDMxYjkwOGU3YTJkMmIxMjAxYmZkODQxNDM5YWIwNTBjYzRlOWMwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.KuMhUXQZzSq7uTOkQ4ib-UmjJTcFxcLOCPnG6Vi1ppo"> this file, its basically just the full build

Ya the problem is I don't get perms and idk how to give myself perms with commands. So it's pretty useless to me since every role is locked

PubliclyExecutedPig commented:
> > > OKAY SO I RAN TOOLS AND THAT WORKED. Corpsman spawns fine, so don't need happy to check-
> > > Well. "Fine" - They still need spawnpoints mapped, but, you know they exist in round.
> > 
> > 
> > ya runserver.bat is just for quick testing on items and such, for actual testing i recommend running <img alt="image" width="207" height="45" src="https://private-user-images.githubusercontent.com/152017811/468275089-71083cf5-c9cc-460e-a25b-5f1b9768b87e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI5MTY2ODgsIm5iZiI6MTc1MjkxNjM4OCwicGF0aCI6Ii8xNTIwMTc4MTEvNDY4Mjc1MDg5LTcxMDgzY2Y1LWM5Y2MtNDYwZS1hMjViLTVmMWI5NzY4Yjg3ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcxOVQwOTEzMDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNzY3NzQ0Njk1YWM1NjI2NjFiMmNjYTM1NDMxYjkwOGU3YTJkMmIxMjAxYmZkODQxNDM5YWIwNTBjYzRlOWMwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.KuMhUXQZzSq7uTOkQ4ib-UmjJTcFxcLOCPnG6Vi1ppo"> this file, its basically just the full build
> 
> Ya the problem is I don't get perms and idk how to give myself perms with commands. So it's pretty useless to me since every role is locked

ah i see, in any case you can do "playtime addrole" command, though tools is probably easier if it unlocks everything

KillerTheFlareon commented:
> > > > OKAY SO I RAN TOOLS AND THAT WORKED. Corpsman spawns fine, so don't need happy to check-
> > > > Well. "Fine" - They still need spawnpoints mapped, but, you know they exist in round.
> > > 
> > > 
> > > ya runserver.bat is just for quick testing on items and such, for actual testing i recommend running <img alt="image" width="207" height="45" src="https://private-user-images.githubusercontent.com/152017811/468275089-71083cf5-c9cc-460e-a25b-5f1b9768b87e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTI5MTY2ODgsIm5iZiI6MTc1MjkxNjM4OCwicGF0aCI6Ii8xNTIwMTc4MTEvNDY4Mjc1MDg5LTcxMDgzY2Y1LWM5Y2MtNDYwZS1hMjViLTVmMWI5NzY4Yjg3ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcxOVQwOTEzMDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNzY3NzQ0Njk1YWM1NjI2NjFiMmNjYTM1NDMxYjkwOGU3YTJkMmIxMjAxYmZkODQxNDM5YWIwNTBjYzRlOWMwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.KuMhUXQZzSq7uTOkQ4ib-UmjJTcFxcLOCPnG6Vi1ppo"> this file, its basically just the full build
> > 
> > 
> > Ya the problem is I don't get perms and idk how to give myself perms with commands. So it's pretty useless to me since every role is locked
> 
> ah i see, in any case you can do "playtime addrole" command, though tools is probably easier if it unlocks everything

Tried using that and it unironically bugged out "release"
But yeah tools is simpler, it's just release with everything unlocked it seems.

KillerTheFlareon committed:
last second addition

gives corps a medical roller bed, since, you know. Probably needs that to effectively drag dead bodies lmao.

Conflee commented:
Sprites all look good!

KillerTheFlareon committed:
Salem offered to make sprites

Who am I to say no?

Fasuh committed:
feat(Chemistry): Add solution indicator on corpsman hypospray

KillerTheFlareon committed:
fixes

KillerTheFlareon commented:
updated to dev branch instead of main

mikeysaurus commented:
This looks like a great addition!! Hope to see it ingame soon

KillerTheFlareon commented:
mfw when a pr that does the same thing is merged faster

KillerTheFlareon committed:
Merge branch 'Starlight' into Brigmed-Hypo

KillerTheFlareon committed:
Merge branch 'starlight-dev' into Brigmed-Hypo

KillerTheFlareon commented:
and this is why i hate a duplicate pr being made that creates conflict with a pre-existing pr.. okay, what went wrong here..

KillerTheFlareon committed:
Update glasses.yml

KillerTheFlareon commented:
fixed

KillerTheFlareon commented:
ughhhhh

KillerTheFlareon committed:
Merge branch 'starlight-dev' into Brigmed-Hypo

KillerTheFlareon commented:
RSI errors are not related to my PR.

KillerTheFlareon committed:
Revert "Merge branch 'starlight-dev' into Brigmed-Hypo"

This reverts commit 8360d2516d49495866edcc50d4c664499480c3a2, reversing
changes made to 19361d0062dbd313f786a6ef961835d3ace812c0.

KillerTheFlareon committed:
Revert "Update glasses.yml"

This reverts commit 19361d0062dbd313f786a6ef961835d3ace812c0.

KillerTheFlareon committed:
Revert "Merge branch 'starlight-dev' into Brigmed-Hypo"

This reverts commit 61106bc8f745d6c073c6195c8b176812e886dd4b, reversing
changes made to 99d3d020dcfdd8fa13b42f4b7de9ab53c1d1d9cd.

KillerTheFlareon committed:
Revert "Merge branch 'Starlight' into Brigmed-Hypo"

This reverts commit 99d3d020dcfdd8fa13b42f4b7de9ab53c1d1d9cd, reversing
changes made to 0b1f21e53332d098268a6d0a0bd3a966eb41d2cb.

KillerTheFlareon commented:
ah i've fucking bungled it

KillerTheFlareon commented:
https://github.com/ss14Starlight/space-station-14/pull/1125
